woocommerce/woocommerce-ios
# REST API: UI for magic link screen

Part of #8918 
Please also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/749.


## Description

This PR adds the UI for the magic link screen of the Jetpack setup flow.
To minimize the effort, I'm reusing `MailPresenter` from `WordPressAuthenticator`.

## Testing instructions

Prerequisite: Make sure that you have access to a site with WooCommerce and no Jetpack.
- Log out of the app or skip onboarding if needed.
- Tap "Enter your site address" on the prologue screen and proceed with the address of your test site.
- Log in with the credentials of an admin.
- When the login completes, tap the Jetback benefit banner at the bottom of the Dashboard screen.
- Tap Log In on the benefit modal.
- Enter the email address of a password-less WPCom account on the presented WPCom email login screen.
- Notice that the magic link screen is displayed. 
- Tap on the Open Mail button. If your device has no mail app, an alert should be displayed saying an email is sent you; if you have only Mail app, the app should then be opened. If you have more than one mail app, a selector should be displayed to let you choose an app.

## Screenshots

<img src="https://user-images.githubusercontent.com/5533851/222385721-90fdf3f0-6ded-4d01-9373-a1c4f791e021.png" width=320 /> 
<img src="https://user-images.githubusercontent.com/5533851/222385736-24d4c507-51c9-40c9-aacf-9e070e764637.png" width=320 />


---
